src/Storages/ObjectStorage/S3/Configuration.cpp

```diff
@@ -192,6 +253,14 @@ void StorageS3Configuration::fromNamedCollection(const NamedCollection & collect
 
 void StorageS3Configuration::fromAST(ASTs & args, ContextPtr context, bool with_structure)
 {
+    /*
+     * Calls to `extractNamedArgumentAndRemoveFromList` need to happen before count is determined:
+     * `size_t count = StorageURL::evalArgsAndCollectHeaders`
+     * This is because extractNamedArgumentAndRemoveFromList alters the list of arguments
+     * */
+    partition_strategy = extractNamedArgumentAndRemoveFromList<std::string>(args, "partition_strategy").value_or("auto");
```

Comment:
Workaround to avoid messing with "if count ==, then followin arguments can be supported"